core/Archive.php

```diff
@@ -800,10 +801,12 @@ private function prepareArchive(array $archiveGroups, Site $site, Period $period
             $doneFlag = $this->getDoneStringForPlugin($plugin, $idSites);
             $this->initializeArchiveIdCache($doneFlag);
 
-            $idArchive = $archiveLoader->prepareArchive($plugin);
+            $idArchives = $archiveLoader->prepareArchive($plugin);
 
-            if ($idArchive) {
-                $this->idarchives[$doneFlag][$periodString][] = $idArchive;
+            if (!empty($idArchives)) {
+                foreach ($idArchives as $idArchive) {
+                    $this->idarchives[$doneFlag][$periodString][] = $idArchive;
```

Comment:
note to myself: do we need to know which idarchives contain only plugin archive etc

Reply:
Shouldn't need to know for requesting data

Reply:
btw a comment explaining why we might get multiple idArchives and which one will be used for what might be good?